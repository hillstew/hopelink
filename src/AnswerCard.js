import React, { Component } from "react";

class AnswerCard extends Component {
  constructor(props) {
    super(props);
  }

  createAnswerPackage = () => {
    let answer = {
      question: this.props.techQuestions[this.props.currentQuestionIndex],
      answer: this.props.currentAnswer
    };
    this.props.updateSavedAnswers(answer);
  };

  render() {
    const { question, link, video } = this.props.techQuestions[
      this.props.currentQuestionIndex
    ];
    return (
      <div className="answerCard-div">
        {this.props.showSavedAnswers === false && <h2>{question}</h2>}
        {this.props.showSavedAnswers && <h2>{this.props.currentQuestion}</h2>}
        {this.props.isSubmitted && (
          <button onClick={this.props.showQuestion}>Show me a question</button>
        )}
        <p>
          <span>YOUR ANSWER:</span> {this.props.currentAnswer}
        </p>
        <div>
          <h3>RESOURCES</h3>
          <a href={link} target="_blank" rel="noopener noreferrer">
            Article about question
          </a>
          <a href={video} target="_blank" rel="noopener noreferrer">
            Video about question
          </a>
        </div>
        {this.props.isSubmitted && (
          <button onClick={this.createAnswerPackage}>Save Answer</button>
        )}
      </div>
    );
  }
}

export default AnswerCard;
