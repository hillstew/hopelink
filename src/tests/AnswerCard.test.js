import React from "react";
import { shallow } from "enzyme";
import AnswerCard from "../AnswerCard";

let mockAnswer = "The answer is";
let mockQuestion = [{
  category: "javascipt",
  link: "https://hacks.mozilla.org/2015/05/es6-in-depth-destructuring/",
  question: "What is object and array destructuring?",
  video: "https://www.youtube.com/watch?v=RJaRRS27100"
}];

describe("AnswerCard", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <AnswerCard currentAnswer={mockAnswer} currentQuestion={mockQuestion} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
