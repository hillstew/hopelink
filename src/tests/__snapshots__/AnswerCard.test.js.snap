// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerCard should match snapshot 1`] = `
<div
  className="answerCard-div"
>
  <h2>
    <Component />
  </h2>
  <button>
    Show me a question
  </button>
  <p>
    <span>
      YOUR ANSWER:
    </span>
     
    The answer is
  </p>
  <div>
    <h3>
      RESOURCES
    </h3>
    <a
      href="https://www.sitepoint.com/demystifying-javascript-variable-scope-hoisting/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Article about question
    </a>
    <a
      href="https://www.youtube.com/watch?v=iJKkZA215tQ&t=244s"
      rel="noopener noreferrer"
      target="_blank"
    >
      Video about question
    </a>
  </div>
  <button
    className="answerCard-save-answer"
    onClick={[Function]}
  >
    Save Answer
  </button>
</div>
`;
